pymeasure/instruments/keithley/keithleyDAQ6510.py

```diff
@@ -0,0 +1,286 @@
+        available ranges are 100E-3 V to 1000 V. If the measurement function is AC, available
+        ranges are 100E-3 to 750 V. Auto-range is disabled when this property is set. """,
+        validator=truncated_range,
+        values=[100E-3, 1000]
+    )
+
+    voltage_nplc = Instrument.control(
+        ":SENS:VOLT:NPLC?", ":SENS:VOLT:NPLC %g",
+        """ Control the number of power line cycles (NPLC) for the DC voltage measurements,
+        which sets the integration period and measurement speed.
+        Takes values from 5E-4 to 15 (60 Hz) or 12 (50 Hz or 400 Hz).
+        The smallest value is the shortest time, and results in the fastest reading rate,
+        but increases the reading noise and decreases the number of usable digits.
+        The largest value is the longest time, and results in the lowest reading rate,
+        but increases the number of usable digits. """
+    )
+
+    ####################
+    # Resistance (Ohm) #
+    ####################
+
+    resistance = Instrument.measurement(
+        ":READ?",
+        """ Measure the resistance in Ohms, if configured for this reading.
+        """
+    )
+
+    resistance_range = Instrument.control(
+        ":SENS:RES:RANG?", ":SENS:RES:RANG:AUTO 0;:SENS:RES:RANG %g",
+        """ Control the resistance range in Ohms. If the measurement function is 2-wire,
+        the available ranges are 10 to 100E6 Ohms. If the measurement function is
+        4-wire resistance with offset compensation off, the available ranges
+        are 1 to 100E6 Ohms. If the measurement function is 4-wire resistance
+        with offset compensation on, the available ranges are 1 to 10E3 Ohms.
+        Auto-range is disabled when this property is set. """,
+        validator=truncated_range,
+        values=[1, 100E6]
+    )
+
+    offset_compensated = Instrument.control(
+        ":SENS:FRES:OCOM?", ":SENS:FRES:OCOM %s",
+        """ Control if offset compensation is used or not.
+        Valid values are OFF, ON, and AUTO. """,
+        validator=strict_discrete_set,
+        values=["OFF", "ON", "AUTO"],
+        map_values=False
+    )
+
+    resistance_nplc = Instrument.control(
+        ":SENS:RES:NPLC?", ":SENS:RES:NPLC %g",
+        """ Control the number of power line cycles (NPLC) for the 2-wire resistance measurements,
+        which sets the integration period and measurement speed.
+        Takes values from 5E-4 to 15 (60 Hz) or 12 (50 Hz or 400 Hz).
+        The smallest value is the shortest time, and results in the fastest reading rate,
+        but increases the reading noise and decreases the number of usable digits.
+        The largest value is the longest time, and results in the lowest reading rate,
+        but increases the number of usable digits. """
+    )
+
+    ####################
+    # Methods        #
+    ####################
+
+    def measure_resistance(self, nplc=1, resistance=100e6, auto_range=True):
+        """ Configures the measurement of resistance.
```

Comment:
Docstrings should be (according to [PEP 257](https://peps.python.org/pep-0257/)) in imperative mode:
```suggestion
        """ Configure the measurement of resistance.
```

Equally below.

Reply:
Changed all that I spotted